dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java

```diff
@@ -196,9 +189,40 @@ public int weeksInYear( int year )
     @Override
     public int isoWeek( DateTimeUnit dateTimeUnit )
     {
-        DateTime dateTime = toIso( dateTimeUnit )
-            .toJodaDateTime( ISOChronology.getInstance( DateTimeZone.getDefault() ) );
-        return dateTime.getWeekOfWeekyear();
+
+        if (dateTimeUnit.getYear() > START_ISO.getYear()) {
```

Comment:
Please correct the code style to apply to the DHIS 2 style with brackets on new line, spaces after ( ).